roles/zabbix_agent/tasks/tlspsk_auto.yml

```diff
@@ -3,75 +3,7 @@
   when: (zabbix_agent_os_family != "Windows") or (zabbix_agent_docker | bool)
 
 - include_tasks: tlspsk_auto_windows.yml
-  when: (zabbix_agent_os_family == "Windows")
-
-- name: AutoPSK | read existing TLS PSK file
-  slurp:
-    src: "{{ zabbix_agent_tlspskfile }}"
-  register: zabbix_agent_tlspsk_base64
-  become: true
-  when: zabbix_agent_tlspskcheck.stat.exists
-  no_log: "{{ ansible_verbosity < 3 }}"
-
-- name: AutoPSK | Save existing TLS PSK secret
-  set_fact:
-    zabbix_agent_tlspsk_read: "{{ zabbix_agent_tlspsk_base64['content'] | b64decode | trim }}"
-  when: zabbix_agent_tlspskcheck.stat.exists
-  no_log: "{{ ansible_verbosity < 3 }}"
-
-- name: AutoPSK | Use existing TLS PSK secret
-  set_fact:
-    zabbix_agent_tlspsk_secret: "{{ zabbix_agent_tlspsk_read }}"
-  when: zabbix_agent_tlspskcheck.stat.exists and zabbix_agent_tlspsk_read|length >= 32
-  no_log: "{{ ansible_verbosity < 3 }}"
-
-- name: AutoPSK | Generate new TLS PSK secret
-  set_fact:
-    zabbix_agent_tlspsk_secret: "{{ lookup('password', '/dev/null chars=hexdigits length=64') }}"
-  when: not zabbix_agent_tlspskcheck.stat.exists or zabbix_agent_tlspsk_read|length < 32
-  no_log: "{{ ansible_verbosity < 3 }}"
-
-- name: AutoPSK | Read existing TLS PSK identity file
-  slurp:
-    src: "{{ zabbix_agent_tlspskidentity_file }}"
-  register: zabbix_agent_tlspskidentity_base64
-  become: true
-  when: zabbix_agent_tlspskidentity_check.stat.exists
-  no_log: "{{ ansible_verbosity < 3 }}"
-
-- name: AutoPSK | Use existing TLS PSK identity
-  set_fact:
-    zabbix_agent_tlspskidentity: "{{ zabbix_agent_tlspskidentity_base64['content'] | b64decode | trim }}"
-  when: zabbix_agent_tlspskidentity_check.stat.exists
-  no_log: "{{ ansible_verbosity < 3 }}"
-
-- name: AutoPSK | Generate new TLS PSK identity
-  set_fact:
-    zabbix_agent_tlspskidentity: >-
-      {{
-        zabbix_agent_visible_hostname
-        | default(((zabbix_agent2 == True) | ternary(zabbix_agent2_hostname, zabbix_agent_hostname)))
-        + '_'
-        + lookup('password', '/dev/null chars=hexdigits length=4')
-      }}
-  when: not zabbix_agent_tlspskidentity_check.stat.exists
-  no_log: "{{ ansible_verbosity < 3 }}"
-
-- name: AutoPSK | Template TLS PSK identity in file
-  copy:
-    dest: "{{ zabbix_agent_tlspskidentity_file }}"
-    content: "{{ zabbix_agent_tlspskidentity }}"
-    owner: zabbix
-    group: zabbix
-    mode: 0400
-  become: true
-  when:
-    - zabbix_agent_tlspskidentity_file is defined
-    - zabbix_agent_tlspskidentity is defined
-  notify:
-    - restart zabbix-agent
-    - restart win zabbix agent
-    - restart mac zabbix agent
+  when: (zabbix_agent_os_family == "Windows") or (zabbix_agent_docker | bool)
```

Comment:
Correct me if I am wrong, but shouldn't this be an AND condition?

Reply:
Good find. That should not be there. I did agent2 to but forgot to remove or condition on agentd